src/util/atomicbool.go

```diff
@@ -1,32 +1,35 @@
 package util
 
-import "sync"
+import (
+	"sync/atomic"
+)
 
 // AtomicBool is a boxed-class that provides synchronized access to the
 // underlying boolean value
 type AtomicBool struct {
-	mutex sync.Mutex
-	state bool
+	state int32 // "1" is true, "0" is false
 }
 
 // NewAtomicBool returns a new AtomicBool
 func NewAtomicBool(initialState bool) *AtomicBool {
-	return &AtomicBool{
-		mutex: sync.Mutex{},
-		state: initialState}
+	var state int32 = 0
+	if initialState == true {
+		state = 1
+	}
+	return &AtomicBool{state: state}
 }
 
 // Get returns the current boolean value synchronously
 func (a *AtomicBool) Get() bool {
-	a.mutex.Lock()
-	defer a.mutex.Unlock()
-	return a.state
+	return atomic.LoadInt32(&a.state) == 1
 }
 
 // Set updates the boolean value synchronously
 func (a *AtomicBool) Set(newState bool) bool {
-	a.mutex.Lock()
-	defer a.mutex.Unlock()
-	a.state = newState
-	return a.state
+	var state int32 = 0
```

Comment:
Can be simplified:

```go
var state in32
if newState {
    state = 1
}
```

Reply:
Yes, we can simplify this. But in this particular case I prefer to explicitly set variable to `0`, so that it would be more obvious that `"1" is true` and `"0" is false`

Reply:
I see your point, but golint complains that we should remove `= 0`, so can you change the code as @leonklingele suggested?

```
src/util/atomicbool.go:15:20: should drop = 0 from declaration of var state; it is the zero value
src/util/atomicbool.go:29:20: should drop = 0 from declaration of var state; it is the zero value
```